services/mailer.js

```diff
@@ -17,18 +17,42 @@ if (smtpRequiredProps.some(prop => !process.env[prop])) {
 }
 
 // load all the templates as strings
-const templateStrings = {};
-fs.readdir(path.join(__dirname, 'email'), (err, files) => {
-  if (err) {
-    throw err;
-  }
+const templates = {};
+
+// load the temlates per request during development
+if (process.env.NODE_ENV !== 'production') {
+  templates.render = (name, format = 'txt', context) => new Promise((resolve, reject) => {
+    fs.readFile(path.join(__dirname, 'email', [name, format, 'ejs'].join('.')), (err, file) => {
+      if (err) {
+        return reject(err);
+      }
 
-  files.forEach(file => {
-    fs.readFile(path.join(__dirname, 'email', file), 'utf8', (err, data) => {
-      templateStrings[file] = _.template(data);
+      return resolve(_.template(file)(context));
     });
   });
-});
+} else {
+  templates.data = fs.readdirSync(path.join(__dirname, 'email')).reduce((data, filename) => {
```

Comment:
Forking the code for development / production in this manner could introduce bugs / maintenance issues that are impossible to detect in staging. 

Please refactor the code that loads/parses the file into a single function that is run in all environment. This way there will be one single method for loading templates regardless of the environment, so bugs will appear in dev/staging. If you'd like to add a convenience method in dev for reloading, the dev only code should trigger that same template loading code to ensure it is getting the files the same way production will.